### PR TITLE
#111 : Reorganize Readme

### PR DESCRIPTION
**Pull request by shivanshbindal9**
Currently, the link to the contributing file is not there.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,131 +1,13 @@
...
-* [API:User group membership](https://www.mediawiki.org/wiki/API:User_group_membership)
-  *  [userrights.py](python/userrights): add and remove user rights
-* [API:Watchlist feed](https://www.mediawiki.org/wiki/API:Watchlist_feed)
-  * [get_my_watchlist_feed.py](python/get_my_watchlist_feed.py): access an RSS feed of your own watchlist
-  * [get_user_watchlist_feed.py](python/get_user_watchlist_feed.py): access an RSS feed of another user's watchlist
-* [API:Options](https://www.mediawiki.org/wiki/API:Options)
-  * [change_user_options.py](python/change_user_options.py): change preferences of current user
-* [API:Emailuser](https://www.mediawiki.org/wiki/API:Emailuser)
-  *  [send_an_email.py](python/send_an_email.py): send an email to user
+<p align="center"><img src="https://upload.wikimedia.org/wikipedia/commons/a/a3/MediaWiki_logo_1.png" height="400px" alt="MediaWiki"/> </p>
```
Comment (srish) [inline README.md]: I think MediaWiki logo needs to be smaller in size and then left aligned.

File: README.md [hunk 2]
```diff
@@ -1,131 +1,13 @@
...
-* [API:Protect](https://www.mediawiki.org/wiki/API:Protect)
-  * [protect.py](python/protect.py): Change the protection level of a given page
-* [API:Upload](https://www.mediawiki.org/wiki/API:Upload)
-  * [upload_file_directly.py](python/upload_file_directly.py): upload a file directly from the system
-  * [upload_file_from_url.py](python/upload_file_from_url.py): upload a file from a URL 
-  * [upload_file_in_chunks.py](python/upload_file_in_chunks.py): upload a file in chunks
-* [API:Pagepropnames](https://www.mediawiki.org/wiki/API:Pagepropnames)
-  * [get_pagepropnames.py](python/get_pagepropnames.py): List all page property names in use on the wiki
+# MediaWiki Action API Code Samples
+The MediaWiki action API is a web service that allows access to some wiki-features like authentication, page operations, and search. It can provide Meta information about the wiki and the logged-in user.
```
Comment (srish) [inline README.md]: MediaWiki Action API to be linked w/ https://www.mediawiki.org/wiki/API:Main_page.

Comment (srish) [reply README.md]: `M` in `Meta information..` to be in lowercase.

File: README.md (hunk 1 above)
Comment (srish) [reply README.md]: `a` in `action` uppercase.

File: README.md [hunk 3]
```diff
@@ -1,131 +1,13 @@
...
-  * [geosearch.py](python/geosearch.py): search for pages nearby
-  * [geoimagesearch.py](python/geoimagesearch.py): search for pages nearby with images
-  * [geocoordinates.py](python/geocoordinates.py): obtain coordinates for wiki pages nearby
-* [API:Opensearch](https://www.mediawiki.org/wiki/API:Opensearch)
-  * [opensearch.py](python/opensearch.py): fetch results in an opensearch format
-* [API:Prefixsearch](https://www.mediawiki.org/wiki/API:Prefixsearch)
-  * [prefixsearch.py](python/prefixsearch.py): perform a prefix search for page titles
-* [API:Languagesearch](https://www.mediawiki.org/wiki/API:Languagesearch)
-  * [languagesearch.py](python/languagesearch.py): search for a language 
+The code snippets in several programming languages are developed to assist developers for easy use of MediaWiki action API's.
```
Comment (srish) [inline README.md]: Rephrase to "Code snippets in several programming languages are developed to assist developers for easy use of various modules of the MediaWiki Action API."

File: README.md [hunk 4]
```diff
@@ -1,131 +1,13 @@
...
 
-### Basics
-* [API:Parameter information](https://www.mediawiki.org/wiki/API:Parameter_information)
-  * [paraminfo.py](python/paraminfo.py): get information about another action API module and its parameters
+Currently, the code snippets are present in  ```Python``` and ```Javascript```. The detailed documentation and code snippets of various API modules present in these languages can be found at
```
Comment (srish) [inline README.md]: Rephrase to "Currently, they are available in programming languages `Python` and `Javascript`: "

File: README.md [hunk 5]
```diff
@@ -137,8 +19,9 @@ Demo app that uses [prop=images](https://www.mediawiki.org/wiki/API:Images) modu
 * [User Contributions](python/demos/UserContributions)
 A sample app that uses MediaWiki Action [API:Usercontribs](https://www.mediawiki.org/wiki/API:Usercontribs) allows you to see the latest top 50 edits made by a user. This app uses Flask Framework as backend.
 
-### Installation
```
Comment (srish) [inline README.md]: For now, let's keep `Installation` & `Contributing code samples` as it is. We will address that in a separate pull request.

File: python/README.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
+# Code Snippets : Python
```
Comment (srish) [inline python/README.md]: Let's keep the heading text as "Python" only.

File: python/README.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+Code snippets in Python demonstrating how to use various modules of the [MediaWiki Action API](https://www.mediawiki.org/wiki/API:Main_page)
```
Comment (srish) [inline python/README.md]: Also add the demo apps in here.

File: README.md [hunk 8]
```diff
@@ -1,131 +1,13 @@
...
+The code snippets in several programming languages are developed to assist developers for easy use of MediaWiki action API's.
 
-### Basics
-* [API:Parameter information](https://www.mediawiki.org/wiki/API:Parameter_information)
-  * [paraminfo.py](python/paraminfo.py): get information about another action API module and its parameters
+Currently, the code snippets are present in  ```Python``` and ```Javascript```. The detailed documentation and code snippets of various API modules present in these languages can be found at
+* [Python](python/)
+* [Javascript](javascript/)
 
 ### Demo apps
```
Comment (srish) [inline README.md]: We can make a bullet point around "Demo apps", don't need a section heading.

File: javascript/README.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
+# Code Snippets : Javascript
```
Comment (srish) [inline javascript/README.md]: Keep the heading as "Javascript".  Also, I see that you've added links to the javascript files that we are currently missing in the project folder. May be remove them for now.

File: python/README.md (hunk 7 above)
Comment (shivanshbindal9) [reply python/README.md]: I was unsure whether I needed to add Demo Apps in the README of python or just include "demo apps" in the text. So, I added both.

**Comment by shivanshbindal9** [comment]
Thanks a lot for review @srish !! I have made all the changes that you suggested.
Should I decrease the size of MediaWiki logo more or is it fine?

**Pull request merged**